File: solutions/2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.js
function closestTarget(W, T, s) {
  let a = Number.MAX_SAFE_INTEGER, n = W.length;
  for (let i = 0; i < n; i++) {
    if (W[i] === T) {
      const d = Math.abs(i - s);
      a = Math.min(a, Math.min(d, n - d));
    }
  }
  return a === Number.MAX_SAFE_INTEGER ? -1 : a;
}

File: solutions/2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.go
package main

func closestTarget(W []string, T string, s int) int {
    a, n := int(^uint(0)>>1), len(W)
    for i := 0; i < n; i++ {
        if W[i] == T {
            d := i - s
            if d < 0 {
                d = -d
            }
            if d > n-d {
                d = n - d
            }
            if d < a {
                a = d
            }
        }
    }
    if a == int(^uint(0)>>1) {
        return -1
    }
    return a
}